### DISCUSSION TITLE
Winmerge cannot properly compare some files with different formating and filenames

### DISCUSSION THREAD

**Discussion by Question2** [WinMerge/winmerge#2228] [Q&A]
Im trying to compare files with different filenames and different text formatting. But Winmerge doesnt appear to be able to do this properly. I am using "compare non-horizontally" because they have different filenames, and that seems to be the only way to get winmerge to compare them.

As an example, i have 3 files i am trying to compare. Files 1 and 2 are similar, and winmerge can compare them without any issues. File 3 is very different, and only contains a portion of the text that files 1 and 2 have. Winmerge doesnt appear to be able to handle that, especially if the text is out of order.

Heres an example of what i mean : 

Files 1 and 2 : 
`
test text

placeholder text
`

File 3 : 

`
placeholder text

test text
`

Winmerge doesn't know how to handle a situation like this when the order of text is different.

A screenshot to illustrate what i mean : 

![image](https://github.com/WinMerge/winmerge/assets/1191378/183e6c4b-5f19-4a71-82f0-8d8147224a60)

In this kind of situation, is there anything i can do with winmerge to compare all 3 files properly? Because winmerge seems to get confused if the file contents are out of order.

I also keep getting errors complaining about "different carriage returns" even though all the files are using UTF-8 encoding. What is this about?

**Comment by sdottaka** [comment]
It may be possible to make it a little easier to see by setting up Synchronization Points. Below is a demo gif animation.

https://github.com/WinMerge/winmerge/issues/668#issuecomment-792812030

If that doesn't work, could you please attach those three files?

> I also keep getting errors complaining about "different carriage returns" even though all the files are using UTF-8 encoding. What is this about?

It's not about the encoding of the files, it's about the EOL being different. There are three types of EOL: Windows (CR+LF), Unix (LF), and Mac (CR).
